// constants/types.js
module.exports = {
  SET_USER_OBJECT: 'SET_USER_OBJECT',
  REMOVE_USER_OBJECT: 'REMOVE_USER_OBJECT',
  SAVE_REGISTER_OBJECT: 'SAVE_REGISTRATION_OBJECT',
  SET_LOGIN_ERROR: 'SET_LOGIN_ERROR',
  SET_REGISTER_ERROR: 'SET_REGISTRATION_ERROR',
  SET_HOME_VIEW: 'SET_HOME_VIEW',
  SET_HOME_BINDING: 'SET_HOME_BINDING',
  SET_MAIN_INDEX: 'SET_MAIN_INDEX',
  SET_MAIN_TITLE: 'SET_MAIN_TITLE',
  SET_PLAYING_TITLE: 'SET_PLAYING_TITLE',
  SET_PLAYING_ARTIST: 'SET_PLAYING_ARTIST',
  SET_PLAYING_SOURCE: 'SET_PLAYING_SOURCE',
  SET_PLAYING_SHUFFLE: 'SET_PLAYING_SHUFFLE',
  SET_PLAYING_REPEAT: 'SET_PLAYING_REPEAT',
  SET_VIDEO_SOURCE: 'SET_VIDEO_SOURCE',
  TOGGLE_VIDEO_PLAY: 'TOGGLE_VIDEO_PLAY',
  STOP_VIDEO: 'STOP_VIDEO',
  PLAY_VIDEO: 'PLAY_VIDEO',
  PAUSE_VIDEO: 'PAUSE_VIDEO',
  SET_VIDEO_CURRENT_TIME: 'SET_VIDEO_CURRENT_TIME',
  SET_VIDEO_DURATION: 'SET_VIDEO_DURATION',
  SET_VIDEO_COMPONENT: 'SET_VIDEO_COMPONENT',
  SET_VIDEO_REPEAT: 'SET_VIDEO_REPEAT',
  REQUEST_NEW_SONGS: 'REQUEST_NEW_SONGS',
  RECEIVE_NEW_SONGS: 'RECEIVE_NEW_SONGS',
  REQUEST_ALL_NEW_SONGS: 'REQUEST_ALL_NEW_SONGS',
  RECEIVE_ALL_NEW_SONGS: 'RECEIVE_ALL_NEW_SONGS',
  NEW_SONGS_ERROR: 'NEW_SONGS_ERROR',
  ALL_NEW_SONGS_ERROR: 'ALL_NEW_SONGS_ERROR',
  REQUEST_POPULAR_SONGS: 'REQUEST_POPULAR_SONGS',
  RECEIVE_POPULAR_SONGS: 'RECEIVE_POPULAR_SONGS',
  REQUEST_ALL_POPULAR_SONGS: 'REQUEST_ALL_POPULAR_SONGS',
  RECEIVE_ALL_POPULAR_SONGS: 'RECEIVE_ALL_POPULAR_SONGS',
  POPULAR_SONGS_ERROR: 'POPULAR_SONGS_ERROR',
  ALL_POPULAR_SONGS_ERROR: 'ALL_POPULAR_SONGS_ERROR',
  REQUEST_BANNER_LINKS: 'REQUEST_BANNER_LINKS',
  RECEIVE_BANNER_LINKS: 'RECEIVE_BANNER_LINKS',
  BANNER_LINKS_ERROR: 'BANNER_LINKS_ERROR',
  START_LOADING_FIRST_BANNER: 'START_LOADING_FIRST_BANNER',
  FINISH_LOADING_FIRST_BANNER: 'FINISH_LOADING_FIRST_BANNER',
  START_LOADING_SECOND_BANNER: 'START_LOADING_SECOND_BANNER',
  FINISH_LOADING_SECOND_BANNER: 'FINISH_LOADING_SECOND_BANNER',

  TRACKS: 0,
  ALBUMS: 1,
  ARTISTS: 2,
  GENRES: 3,
}
