// styles/index.js
import EStyleSheet from 'react-native-extended-stylesheet';
import { Platform } from 'react-native';
const viewHeight = Platform.OS.toLowerCase() === 'ios' ? 96 : 100
const margin = 100 - viewHeight

EStyleSheet.build({
  navBarMargin: 10,
  fullWidth: '100%',
  purchaseBannerHeight: '30%',
  songBannerHeight: '40%',
  minusSongBannerHeight: '36%',
  minusPurchaseBannerHeight: '88%',
  navBarHeight: '12%',
  minusNavBarHeight: '88%',
  minusNavBarAndBannerHeight: '70%',
  milkBackground: '#f2f2f2'
});

const styles = EStyleSheet.create({
  allScreen: {
    flex: 1
  },
  full: {
    width: '100%',
    height: `${viewHeight}%`,
    marginTop: `${margin}%`
  },
  loadingScreen: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    opacity: 1,
    backgroundColor: 'rgba(0,0,0,0.8)'
  },
  scrollContainer: {
    flex: 1,
  },
  filterContainer: {
    height: '8%',
    backgroundColor: '$milkBackground'
  },
  drawerHeading: {
    paddingLeft: '6%',
    flex: 0.122,
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: 'rgba(0,0,0,0.8)'
  },
  drawerMarginTop: {
    flex: 0.09,
    backgroundColor: 'white'

  },
  drawerHeadingLogo: {
    marginRight: '2%',
    height: '10%',
    width: '12%'
  },
  drawerHeadingText: {
    flex: 3,
    color: 'white',
    letterSpacing: '0.3rem'
  },
  drawerContent: {
    flex: 0,
    backgroundColor: 'rgba(100,0,0,0.5)'
  },
  drawerBottom: {
    flex: 0.6,
    backgroundColor: 'white'
  },
  drawerList: {
    height: '8%',
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center'
  },
  drawerIcon: {
    color: 'black',
    fontSize: 25,
    paddingLeft: '6%',
    paddingRight: '8%'
  },
  drawerFooter: {
    flex: 3.3,
    paddingTop: '5%',
    paddingLeft: '5%',
    backgroundColor: 'white',
  },
  mainNavBar: {
    height: '$navBarHeight',
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    backgroundColor: '$milkBackground'
  },
  minusNavBar: {
    height: `${100 - (parseInt(EStyleSheet.value('$navBarMargin')) + 8) - 12}%`,
    paddingBottom: 100
  },
  mainNavBarView: {
    flex: 1,
    flexDirection: 'row'
  },
  mainNavBarMargin: {
    paddingTop: '4%',
    paddingLeft: '2%',
    paddingRight: '2%'
  },
  excludeNavBar: {
    height: `${(100 - EStyleSheet.value('$navBarMargin'))}%`,
  },
  padRight: {
    paddingRight: '5%'
  },
  navBarMenuIcon: {
    fontSize: 22,
    paddingLeft: '3%',
    paddingRight: '4%',
    color: 'red'
  },
  navBarText: {
    fontWeight: 'bold',
    fontSize: 17
  },
  container: {
    flex: 1,
    width: undefined,
    height: undefined,
    justifyContent: 'flex-end',
    backgroundColor: 'transparent',
  },
  imageContainer: {
    flex: 1,
    width: undefined,
    height: undefined,
    justifyContent: 'center',
    backgroundColor: 'transparent',
  }
});

styles.homeBackground = require('images/home_bg.jpg')
styles.logoImage = require('images/logo.png')
styles.musicImage = require('images/music.png')
module.exports = styles;
