// styles/index.js
import EStyleSheet from 'react-native-extended-stylesheet';
import { Platform } from 'react-native';
const viewHeight = Platform.OS.toLowerCase() === 'ios' ? 96 : 100
const margin = 100 - viewHeight

EStyleSheet.build({
  navBarMargin: 10,
  fullWidth: '100%',
  purchaseBannerHeight: '30%',
  songBannerHeight: '40%',
  minusSongBannerHeight: '36%',
  minusPurchaseBannerHeight: '88%',
  navBarHeight: '12%',
  minusNavBarHeight: '88%',
  minusNavBarAndBannerHeight: '70%',
  milkBackground: '#f2f2f2'
});

const styles = EStyleSheet.create({
  allScreen: {
    flex: 1
  },
  full: {
    width: '100%',
    height: `${viewHeight}%`,
    marginTop: `${margin}%`
  },
  loadingScreen: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    opacity: 1,
    backgroundColor: 'rgba(0,0,0,0.8)'
  },
  scrollContainer: {
    flex: 1,
  },
  filterContainer: {
    height: '8%',
    backgroundColor: '$milkBackground'
  },
  drawerHeading: {
    paddingLeft: '6%',
    flex: 0.122,
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: 'rgba(0,0,0,0.8)'
  },
  drawerMarginTop: {
    flex: 0.09,
    backgroundColor: 'white'

  },
  drawerHeadingLogo: {
    marginRight: '2%',
    height: '10%',
    width: '12%'
  },
  drawerHeadingText: {
    flex: 3,
    color: 'white',
    letterSpacing: '0.3rem'
  },
  drawerContent: {
    flex: 0,
    backgroundColor: 'rgba(100,0,0,0.5)'
  },
  drawerBottom: {
    flex: 0.6,
    backgroundColor: 'white'
  },
  drawerList: {
    height: '8%',
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center'
  },
  drawerIcon: {
    color: 'black',
    fontSize: 25,
    paddingLeft: '6%',
    paddingRight: '8%'
  },
  drawerFooter: {
    flex: 3.3,
    paddingTop: '5%',
    paddingLeft: '5%',
    backgroundColor: 'white',
  },
  mainNavBar: {
    height: '$navBarHeight',
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    backgroundColor: '$milkBackground'
  },
  minusNavBar: {
    height: `${100 - (parseInt(EStyleSheet.value('$navBarMargin')) + 8) - 12}%`,
    // height: 450,
    paddingBottom: 100
  },
  mainNavBarView: {
    flex: 1,
    flexDirection: 'row'
  },
  mainNavBarMargin: {
    paddingTop: '4%',
    paddingLeft: '2%',
    paddingRight: '2%'
  },
  excludeNavBar: {
    height: `${(100 - EStyleSheet.value('$navBarMargin'))}%`,
  },
  musicHome: {
    marginTop: '2%',
    paddingBottom: '3%'
  },
  musicHomeHeaderView: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    maxHeight: '8%',
    backgroundColor: 'rgba(100,100,0,0.05)',
    paddingRight: '7%'
  },
  purchaseMoreView: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-end',
    maxHeight: '6%',
    paddingRight: '7%',
    marginBottom: '2%',
  },
  musicHomeMore: {
    borderWidth: 2,
    borderRadius: 5,
    paddingTop: 8,
    paddingBottom: 8,
    paddingRight: 20,
    paddingLeft: 20,
  },
  musicHomeHeader: {
    marginLeft: '4%',
    marginRight: '2%',
    fontSize: 16,
    fontWeight: 'bold',
    color: 'black',
  },
  purchaseBanner: {
    width: '$fullWidth',
    height: '$purchaseBannerHeight'
  },
  songBanner: {
    width: '$fullWidth',
    height: '$songBannerHeight'
  },
  songRest: {
    backgroundColor: '$milkBackground',
    height: '$minusSongBannerHeight',
    paddingTop: '3%',
    flex: 1
  },
  songPurchaseHeaderSection: {
    flex: 2.2,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingLeft: '6%',
    paddingRight: '6%'
  },
  buyButton: {
    borderWidth: 1,
    borderColor: 'black',
    backgroundColor: 'white',
    paddingTop: '1.3%',
    paddingBottom: '1.3%',
    paddingLeft: '3%',
    paddingRight: '3%',
    borderRadius: 5
  },
  purchaseRowKey: {
    fontWeight: 'bold',
    width: '40%'
  },
  songPageTitle: {
    fontSize: '1.4rem',
  },
  songPageArtist: {
    fontSize: '0.9rem',
  },
  songDetails: {
    flex: 4,
    marginLeft: '6%'
  },
  songPurchaseRow: {
    flex: 2
  },
  minusBanner: {
    height: '50%',
    paddingBottom: '4%'
  },
  purchaseSection: {
    borderWidth: 1,
    borderColor: 'black',
    width: '100%',
    height: '7%'
  },
  purchaseSectionInner: {
    margin: 1,
    flex: 1,
    justifyContent: 'center',
    borderWidth: 1,
    borderColor: 'black'
  },
  padRight: {
    paddingRight: '5%'
  },
  purchaseHotSongs: {
    width: '100%',
    height: '10%'
  },
  musicListItem: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    height: '9%',
    paddingBottom: '1%',
  },
  musicListView: {
    paddingTop: '3%',
    paddingBottom: '3%',
  },
  listingSelection: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-end',
    height: '8%',
    marginRight: '3%'
  },
  purchaseListView: {
    paddingTop: '2%',
    minHeight: '20%'
  },
  pickerWidth: {
    width: '40%'
  },
  musicListInfoContainer: {
    flex: 1,
    justifyContent: 'center'
  },
  musicListInfoArtist: {
    fontSize: 11,
    fontFamily: 'Trebuchet MS',
    fontWeight: 'bold'
  },
  musicListItemImage: {
    height: '7%',
    marginLeft: '5%',
    marginRight: '5%',
    width: '14%',
    borderRadius: 4
  },
  player: {
    flex: 1
  },
  playerHeader: {
    flex: 1
  },
  playerImageView: {
    flex: 6
  },
  playerImage: {
    width: undefined,
    height: undefined,
    flex: 1
  },
  playerControl: {
    flex: 2
  },
  playerSongInfo: {
    flex: 1,
    marginLeft: '10%',
    justifyContent: 'center',
    alignItems: 'flex-start'
  },
  playerIcon: {
    fontSize: 35
  },
  playerButtons: {
    flex: 1,
    flexDirection: 'row',
    marginLeft: '10%',
    marginRight: '10%',
    width: '80%',
    marginTop: '4%',
    justifyContent: 'space-between'
  },
  navBarMenuIcon: {
    fontSize: 22,
    paddingLeft: '3%',
    paddingRight: '4%',
    color: 'red'
  },
  navBarText: {
    fontWeight: 'bold',
    fontSize: 17
  },
  controlText: {
    color: 'black',
  },
  button: {
    backgroundColor: 'white',
    borderWidth: 1,
    borderColor: 'black',
    padding: 10,
  },
  container: {
    flex: 1,
    width: undefined,
    height: undefined,
    justifyContent: 'flex-end',
    backgroundColor: 'transparent',
  },
  imageContainer: {
    flex: 1,
    width: undefined,
    height: undefined,
    justifyContent: 'center',
    backgroundColor: 'transparent',
  },
  userComponent: {
    flex: 3.5,
  },
  loginComponent: {
    flex: 5
  },
  regComponent: {
    flex: 11
  },
  logo: {
    flex: 1,
    height: '10%',
    width: '24%'
  },
  center: {
    alignItems: 'center'
  },
  centerWhiteText:{
    color: 'white',
    textAlign: 'center'
  },
  homeText: {
    color: 'white',
    letterSpacing: '0.3rem'
  },
  buttonGroup: {
    marginTop: '3.5%',
  },
  inputContainer: {
    width: '75%'
  },
  inputText: {
    backgroundColor: 'white',
    padding: 8,
    height: '5.5%',
    fontSize: 14,
    width: '75%',
    margin: 4,
    borderRadius: 5
  },
  buttonContainer: {
    width: '75%',
    height: '6.5%',
    backgroundColor: 'green',
    margin: 7,
    padding: 8,
    borderRadius: 5
  },
  addBorder: {
    borderWidth: 3,
    borderColor: 'red'
  },
  bottomText: {
    fontSize: 12,
    letterSpacing: '0.03rem',
    fontWeight: 'bold'
  },
  bottomTextContainer: {
    marginTop: 6,
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
  }
});

styles.homeBackground = require('images/home_bg.jpg')
styles.logoImage = require('images/logo.png')
styles.musicImage = require('images/music.png')
module.exports = styles;
